gourmet/reccard.py

```diff
@@ -120,8 +121,9 @@ def update_recipe (self, recipe):
         self.current_rec = recipe
         if hasattr(self,'recipe_display'):
             self.recipe_display.update_from_database()
-        if self.re is not None and not self.re.window.is_visible():
-            self.re = None
+        if (self.rec_editor is not None and
+            not self.rec_editor.window.is_visible()):
```

Comment:
I'm open to doing checks like this as `self.rec_editor and self.rec_editor.window.is_visible()`, especially if it means that it fits on one line. But I'm not going to push for that if you prefer the specific `is not None` check.

Reply:
Good idea, I was trying to keep everything clear in my mind at the time.

Reply:
The `is not None` check is specifically supported by `mypy` as means to resolve `Optional[A]` into `A`. I am all for it.

Reply:
Okay, let's keep it :)